scripts/tests/core-integration.test.mjs

```diff
@@ -195,253 +190,81 @@ describe("CORE Integration Tests", () => {
-      const dir = path.dirname(relativePath);
-      const basename = path.basename(relativePath);
-
-      let outputPath = relativePath;
+      const basename = path.basename(inputPath);
+      let outputPath = inputPath;
       if (basename === "README.md" && PATH_MAPPINGS.singleFile[dir]) {
         outputPath = PATH_MAPPINGS.singleFile[dir];
       }
 
-      expect(outputPath).toBe("reference/best-practices.md");
-    });
-  });
-
-  describe("Error Handling", () => {
-    it("should handle missing files gracefully", async () => {
-      const missingFile = path.join(docsDir, "nonexistent.md");
-
-      let hasError = false;
-      try {
-        await fs.access(missingFile);
-      } catch {
-        hasError = true;
-      }
-
-      expect(hasError).toBe(true);
-    });
-
-    it("should handle empty content files", async () => {
-      const emptyFile = path.join(docsDir, "empty.md");
-      await fs.writeFile(emptyFile, "", "utf8");
-
-      const content = await fs.readFile(emptyFile, "utf8");
-      expect(content).toBe("");
-
-      const heading = content.match(/#[\s]+(.*)/);
-      expect(heading).toBeNull();
-    });
-
-    it("should handle malformed markdown", async () => {
-      const malformedFile = path.join(docsDir, "malformed.md");
-      const content = "This has no heading\n\nJust content...";
-      await fs.writeFile(malformedFile, content, "utf8");
-
-      const fileContent = await fs.readFile(malformedFile, "utf8");
-      const heading = fileContent.match(/#[\s]+(.*)/);
-
-      expect(heading).toBeNull();
+      expect(outputPath).toBe(expectedOutput);
     });
-  });
 
-  describe("Parallel Processing", () => {
-    it("should handle multiple files processed in parallel", async () => {
-      const files = [
-        "010_user-guide/getting-started.md",
-        "010_user-guide/pepr-cli.md",
-        "020_actions/mutate.md",
-        "020_actions/validate.md",
-        "040_tutorials/basic.md",
-      ];
+    it("should handle malformed markdown by returning null heading", () => {
+      const contentWithoutHeading = "This has no heading\n\nJust content...";
+      const contentWithHeading = "# Proper Heading\n\nContent...";
 
-      await Promise.all(
-        files.map(async file => {
-          const filePath = path.join(docsDir, file);
-          await fs.mkdir(path.dirname(filePath), { recursive: true });
-          const content = `# ${path.basename(file, ".md")}\n\nContent for ${file}`;
-          await fs.writeFile(filePath, content, "utf8");
-        }),
-      );
-
-      const results = await Promise.all(
-        files.map(async file => {
-          const filePath = path.join(docsDir, file);
-          const content = await fs.readFile(filePath, "utf8");
-          return { file, content };
-        }),
-      );
+      const noHeading = contentWithoutHeading.match(/#[\s]+(.*)/);
+      const hasHeading = contentWithHeading.match(/#[\s]+(.*)/);
```

Comment:
Do we use this specific regex somewhere in our "production code"? As written here, it seems decoupled from anything that runs during the docs process. How might we change our docs processing code in a way that could cause this test to fail without changing the implementation of this test?